pkg/handlers/default.go

```diff
@@ -30,37 +31,53 @@ func newDefaultHandler(handlerType handlerType) *defaultHandler {
 // utilizing a single output channel. It first tries to identify the input as an archive. If it is an archive,
 // it processes it accordingly; otherwise, it handles the input as non-archive content.
 // The function returns a channel that will receive the extracted data bytes and an error if the initial setup fails.
-func (h *defaultHandler) HandleFile(ctx logContext.Context, input fileReader) (chan []byte, error) {
+func (h *defaultHandler) HandleFile(ctx logContext.Context, input fileReader) chan DataOrErr {
 	// Shared channel for both archive and non-archive content.
-	dataChan := make(chan []byte, defaultBufferSize)
+	dataOrErrChan := make(chan DataOrErr, defaultBufferSize)
 
 	go func() {
-		defer close(dataChan)
+		defer close(dataOrErrChan)
 
 		start := time.Now()
-		err := h.handleNonArchiveContent(ctx, newMimeTypeReaderFromFileReader(input), dataChan)
+		err := h.handleNonArchiveContent(ctx, newMimeTypeReaderFromFileReader(input), dataOrErrChan)
 		if err == nil {
 			h.metrics.incFilesProcessed()
 		}
 
 		// Update the metrics for the file processing and handle errors.
-		h.measureLatencyAndHandleErrors(start, err)
+		h.measureLatencyAndHandleErrors(ctx, start, err, dataOrErrChan)
 	}()
 
-	return dataChan, nil
+	return dataOrErrChan
 }
 
 // measureLatencyAndHandleErrors measures the latency of the file processing and updates the metrics accordingly.
 // It also records errors and timeouts in the metrics.
-func (h *defaultHandler) measureLatencyAndHandleErrors(start time.Time, err error) {
+func (h *defaultHandler) measureLatencyAndHandleErrors(
+	ctx logContext.Context,
+	start time.Time,
+	err error,
+	dataErrChan chan DataOrErr,
+) {
```

Comment:
I'm not super familiar with this area of the codebase, but it's starting to smell like this method is doing a little too much.

Am I reading this right, that we're now writing to the `dataErrChan` instead of just recording metrics? (side nit: make `dataErrChan` a write-only parameter).

Reply:
I agree. I can create a follow-up PR to break this function into smaller parts. You're correct—we need to pass all errors to dataErrChan instead of just logging them and moving on, as we did before.